src/core/DrumSynth.cpp

```diff
@@ -172,37 +171,19 @@ void DrumSynth::GetEnv(int env, const char* sec, const char* key, QString ini)
 
 float DrumSynth::waveform(float ph, int form)
 {
-	float w;
-
-	switch (form)
-	{
-	case 0:
-		w = static_cast<float>(sin(fmod(ph, TwoPi)));
-		break; // sine
-	case 1:
-		w = static_cast<float>(fabs(2.0f * static_cast<float>(sin(fmod(0.5f * ph, TwoPi))) - 1.f));
-		break; // sine^2
-	case 2:
-		while (ph < TwoPi)
-		{
-			ph += TwoPi;
-		}
-		w = 0.6366197f * static_cast<float>(fmod(ph, TwoPi) - 1.f); // tri
-		if (w > 1.f)
-		{
-			w = 2.f - w;
-		}
-		break;
-	case 3:
-		w = ph - TwoPi * static_cast<float>(static_cast<int>(ph / TwoPi)); // saw
-		w = (0.3183098f * w) - 1.f;
-		break;
-	default:
-		w = (sin(fmod(ph, TwoPi)) > 0.0) ? 1.f : -1.f;
-		break; // square
-	}
-
-	return w;
+	// sine
+	if (form == 0) { return std::sin(ph); }
+	// sine^2
+	if (form == 1) { return std::abs(2.f * std::sin(0.5f * ph)) - 1.f; }
+	// sawtooth with range [0, 1], used to generate triangle, sawtooth, and square
+	auto ph_tau = ph * numbers::inv_tau_v<float>;
```

Comment:
```suggestion
	auto ph_tau = ph / numbers::tau_v<float>;
```
I don't really see the need for `inv_tau_v`.

Reply:
The conventional wisdom is that floating point multiplication is much faster than floating point division. Multiplying by a reciprocal is commonly used to take advantage of this. This is why there's lots of `x * 0.5` instead of `x / 2.0` in this repo, for example.

It's not strictly necessary and I'm willing to make your suggested change, but I'd like to clarify my intent in adding `inv_tau` in the first place.

Reply:
If `1 * inv_tau` is faster than `1 / tau`, I would expect the compiler to optimize that. And even if it didn't, I don't think it's worth adding `inv_*` versions for each constant.

Reply:
I'll go ahead and remove it, but I do want to point out that `std::numbers::inv_pi`, `std::numbers::inv_sqrtpi`, and `std::numbers::inv_sqrt3` exist.

Reply:
https://godbolt.org/z/7TMWMTnoa

It appears the compiler can make that optimization, but only if you have both `-O3` *and* `-ffast-math`. Without both flags, it will optimize `x / 2.0` to `x * 0.5`, but it won't do the same for `x / 3.0` or other `constexpr` values.